core/src/main/scala/org/broadinstitute/dsp/workbench/welder/GoogleStorageInterp.scala

```diff
@@ -0,0 +1,74 @@
+package org.broadinstitute.dsp.workbench.welder
+
+import java.time.Instant
+import java.util.concurrent.TimeUnit
+
+import fs2.{Stream, io}
+import cats.implicits._
+import cats.effect.{ContextShift, IO, Timer}
+import _root_.io.chrisdavenport.log4cats.Logger
+import _root_.io.chrisdavenport.linebacker.Linebacker
+import org.broadinstitute.dsde.workbench.google2
+import org.broadinstitute.dsde.workbench.google2.{GcsBlobName, GoogleStorageService, RemoveObjectResult}
+import org.broadinstitute.dsde.workbench.model.{TraceId, WorkbenchEmail}
+import org.broadinstitute.dsde.workbench.model.google.GcsBucketName
+import org.broadinstitute.dsp.workbench.welder.SourceUri.GsPath
+import GoogleStorageAlg._
+
+class GoogleStorageInterp(config: GoogleStorageAlgConfig , googleStorageService: GoogleStorageService[IO])
+                         (implicit logger: Logger[IO], timer: Timer[IO], linerBacker: Linebacker[IO], cs: ContextShift[IO]) extends GoogleStorageAlg {
+  def updateMetadata(gsPath: GsPath, traceId: TraceId, metadata: Map[String, String]): IO[Unit] = googleStorageService.setObjectMetadata(gsPath.bucketName, gsPath.blobName, metadata, Option(traceId)).compile.drain.handleErrorWith {
+    case e: com.google.cloud.storage.StorageException if(e.getCode == 403) =>
+      for {
+        _ <- logger.info(s"$traceId | Fail to update lock due to 403. Going to download the blob and re-upload")
+        bytes <- googleStorageService.getObjectBody(gsPath.bucketName, gsPath.blobName, Some(traceId)).compile.to[Array]
+        _ <- googleStorageService.storeObject(gsPath.bucketName, gsPath.blobName, bytes, "text/plain", metadata, None, Some(traceId)).compile.drain
+      } yield ()
+  }
+
+  def retrieveGcsMetadata(localPath: RelativePath, bucketName: GcsBucketName, blobName: GcsBlobName, traceId: TraceId): IO[Option[GcsMetadata]] = for {
```

Comment:
this tripped me up a bit when i was reading the `acquireLock` function in `ObjectService`. this _is_ retrieving gcs metadata like it says but it's returning a transformed version of that metadata which isn't actually the gcs metadata anymore. what i was looking for in `acquireLock` was a condition for when another user's lock has expired but ended up having to come here to realize that the lock metadata is thrown out if it's expired.

could you rename this function to accurately reflect what it's really doing?